test/functional/test-activity.js

```diff
@@ -116,7 +118,8 @@ describe('Activity getTotalEngagedTime', () => {
       scrollObservable.add(handler);
     });
 
-    fromClassForDoc(ampdoc, 'activity', Activity);
+    markElementScheduledForTesting(fakeWin, 'amp-analytics');
```

Comment:
fromClassForDoc instantiates the service directly. Installing just registers the service. Then activityForDoc calls getElementServiceForDoc, which itself won't try to instantiate the service unless this happens.